javascript-sdk/src/sdk.ts

```diff
@@ -28,38 +27,45 @@ export default class RebuffSdk implements Rebuff {
   private strategies: Record<string, Strategy> | undefined;
   private defaultStrategy: string;
 
-  private openai: {
-    conn: OpenAIApi;
-    model: string;
-  };
-
+  /**
+   * @deprecated Use `RebuffSdk.init` instead.
+   */
   constructor(config: SdkConfig) {
+    // We're keeping this constructor for backwards compatibility. In the future, we can make it private and
+    // simplify this class quite a bit.
```

Comment:
Curious what issues we're running into with backwards compatibility, is it an issue with the test workflows? I get the idea behind deprecating the constructor because we want the init to be the primary method, but we're still using the constructor inside the init so deprecating it doesn't seem to be exactly what we want. I'm wondering if we can fix whatever's causing the backwards compatibility issue.

Reply:
The issues that I've run into is that when I make a breaking change to the JS SDK, I am not able to update the code in `server/` until my PR with the breaking change is merged, and a new NPM version of rebuff is released. In the time between when the breaking change is made and the updates to `server/` is made, anyone who wants to make changes to the SDK and test it locally with the server will have to update the `server/` code to account for the breaking change, either manually or by cherry-picking the commits from the PR to update `server/`.

Another issue is the sort of thing I encountered in #96 where my choice was to have a `javascript-sdk` directory that doesn't compile or a JS API client that doesn't match the server.

It will only be a temporary state that we're calling a deprecated constructor from within `init()`. When I make the PR to update `server/`, I'll make the constructor private and remove the deprecation comment. If you think it would be less confusing, I can remove the deprecated comment now.

Reply:
Makes sense, thanks for the explanation. The deprecated decorator isn't a huge deal to me. I'm fine with merging it as is.